### DISCUSSION TITLE
What is the differene between the clipboard and the primary clipboard?

### DISCUSSION THREAD

**Discussion by aslowwriter** [helix-editor/helix#8592] [Q&A] [answered]
The documentation states:
> :clipboard-yank | Yank main selection into system clipboard.
...
> :primary-clipboard-yank | Yank main selection into system primary clipboard.

It is really unclear to me what the difference is between these two, as both seem to pase the system's clipboard? Can anyone explain to me what the primary clipboard (or the non-primary for that matter) is?

**Comment by kirawi** [comment]
IIRC, primary clipboard is what is generally used for copy-paste operations. The non-primary keyboard can be e.g. the terminal middle-click clipboard. You can see https://github.com/helix-editor/helix/blob/88bc52a57058941e8f56b298b179426023184ac3/helix-view/src/clipboard.rs#L100

**Comment by dead10ck** [comment, marked as answer]
This is a Linux thing. See

https://unix.stackexchange.com/questions/139191/whats-the-difference-between-primary-selection-and-clipboard-buffer
